index.js

```diff
@@ -91,13 +84,14 @@ export default class SideMenu extends React.Component {
     this.onPanResponderTerminate = this.handlePanResponderEnd.bind(this);
 
     this.state = {
-      width: deviceScreen.width,
-      height: deviceScreen.height,
+      width,
+      height,
       openOffsetMenuPercentage,
-      openMenuOffset: deviceScreen.width * openOffsetMenuPercentage,
+      openMenuOffset: width * openOffsetMenuPercentage,
       hiddenMenuOffsetPercentage,
-      hiddenMenuOffset: deviceScreen.width * hiddenMenuOffsetPercentage,
+      hiddenMenuOffset: width * hiddenMenuOffsetPercentage,
       left,
+      barrierForward: width / 4,
```

Comment:
I don't see you update this value anywhere. Why do you want to keep it inside the state?

Reply:
`this.state.barrierForward` is used in `SideMenu.handlePanResponderEnd`, passed to the module level function `shouldOpenMenu`. Before, `shouldOpenMenu` didn't take `barrierForward` as an argument, instead using a value calculated from the module's scope. 

The old module-level `barrierForward` variable is calculated based on `deviceScreen`, which is never updated to account for device rotations. Since this PR removes the module-level `deviceScreen` in favor of getting fresh device dimensions in the component's constructor, I had to change a few other things that were using this value, `shouldOpenMenu` being one of them